src/gravity.js

```diff
@@ -0,0 +1,508 @@
+  // right is the floor
+  right: { up: 'left', down: 'right', left: 'up', right: 'down' },
+  // left is the floor
+  left: { up: 'right', down: 'left', left: 'down', right: 'up' }
+}
+var lr = ['left',  'right'];
+var ud = ['up',  'down'];
+var dirs = ['up', 'down', 'left', 'right'];
+var offsets = {
+  up: [0, -1],
+  down: [0, 1],
+  left: [-1, 0],
+  right: [0, 1],
+}
+
+/* THE MAGIC
+*/
+
+after("onPlayerMoved", function () {
+  if (!active) return;
+  var player = bitsy.player();
+
+  wasStandingOnSomething = isSolid(gravityDir, player.x, player.y);
+
+  // if player is standing on something and has a fallCounter > 0, then they just landed
+  if (wasStandingOnSomething && fallCounter && hackOptions.fallDamage) hackOptions.fallDamage(fallCounter);
+})
+
+before("movePlayer", function () {
+  if (!active) return;
+  var player = bitsy.player();
+
+  wasStandingOnSomething = isSolid(gravityDir, player.x, player.y);
+
+  if (wasStandingOnSomething) {
+    // reset jetpack fuel
+    jumpPower = hackOptions.jumpPower;
+  }
+
+  wasClimbing = isClimbing;
+  isOnClimbableTile = isTileClimbable(player.x, player.y);
+
+  if (wasStandingOnSomething || isClimbing || wasJetpacking || forceGravityDir) {
+
+    // reset fall counters
+    fallCounter = 0;
+    horizontalFallingMoves = 0;
+  } else {
+    // player is falling
+    fallCounter += 1;
+  }
+
+
+
+  if (forceGravityDir && isSolid(forceGravityDir, player.x, player.y)) {
+    // if played was being pushed, and hit a wall in that direction, turn off the push
+    forceGravityDir = undefined;
+  }
+
+});
+
+
+window.advanceGravity = function () {
+  if (!active) return;
+  var player = bitsy.player();
+  // player input something, but could not move.
+
+  // force them up if they are doing that
+  if (forceGravityDir) {
+    reallyMovePlayer(player, forceGravityDir)
+    lastMoveMapped = forceGravityDir;
+    return;
+  }
+
+  if (isOnStandableTile(player)) return;
+
+  // otherwise:force them downward if possible.
+  reallyMovePlayer(player, gravityDir);
+  lastMoveMapped = 'down';
+}
+
+window.movePlayerWithGravity = function (dir, axis, amt) {
+  var player = bitsy.player();
+  if (!active) {
+    player[axis] += amt;
+    return;
+  }
+
+  // if player is being pushed, just push em and skip everything else
+  if (forceGravityDir) {
+    reallyMovePlayer(player, forceGravityDir)
+    lastMoveMapped = forceGravityDir;
+    return;
+  }
+
+  // dir is the arrow pressed. realDir is what that arrow translates to in the current gravity inversion...
+  var realDir = mapDir(dir)
+  var horizontal = realDir === 'left' || realDir === 'right';
+
+  var canJetpack = !!jumpPower;
+  // if set to jetpack false, only let them jump if they are standing on something or used jetpack last move
+  if (!jetpack && !wasStandingOnSomething && !wasJetpacking) canJetpack = false
+
+// reset vars and stuff
+  wasJetpacking = false;
+  isClimbing = false;
+
+  // THE BIG MAMA
```

Comment:
keep in mind the intent for the hacks is other people being able to jump in and play around with them with potentially little to no programming experience; not a strict rule or anything but when leaving comments it's best to try to keep them direct + instructional